experiments/compositions/composition/api/v1alpha1/composition_types.go

```diff
@@ -119,15 +120,27 @@ type CompositionSpec struct {
 	//InputNamespace string     `json:"inputNamespace,omitempty" protobuf:"bytes,5,name=inputNamespace"`
 	//Sinc      Sinc       `json:"sinc,omitempty" protobuf:"bytes,6,name=sinc"`
 
-	Description   string `json:"description,omitempty" protobuf:"bytes,2,name=description"`
-	InputAPIGroup string `json:"inputAPIGroup,omitempty" protobuf:"bytes,3,name=inputAPIGroup"`
+	Description string `json:"description,omitempty"`
+
+	// Use existing KRM API created offline
+	// TODO (barney-s) rename to FacadeAPIGroup,facadeAPIGroup
+	InputAPIGroup string `json:"inputAPIGroup,omitempty"`
+
+	// Bring your own open API spec
```

Comment:
Nice functionality but would it make more sense to have this as a command line tool?

Reply:
Absolutely. Kubebuilder/controller-gen is a simpler one to generate APIs. 

Some think it is useful to do something like Gatekeeper ConstraintTemplates. 
At the moment, i am just accommodating those kinds of users. 
The jury is out.

Reply:
@cheftako - Should we merge this for adding this capability and later align to a simpler schema (custom one as demoed in the call today) ?

Reply:
I think we are littering the CompositionSpec. At some level I don't think we should care where the CRD came from. If we want another way of generating a CRD, lets put that in a dedicated object but leave composition as streamlined and simple as possible.

Reply:
Please separate this functionality into its own CRD. Don't want to conflate the composition lifecycle with the CRD lifecycle.

Reply:
Using `Facade` for the new API for managing CRDs.

Reply:
I have separated the BYO Schema creation functionality into a separate `Facade` CRD.